### PR TITLE
lldbutil: Forward ASan launch info to test inferiors

### PR DESCRIPTION
**Pull request by vedantk**
This is a WIP patch to address ASan failures in some swift lldb tests.

I've created the PR to get additional test coverage, but the real change should happen on llvm.org.

### CODE REVIEW

**Comment by vedantk** [comment]
@swift-ci test

File: lldb/packages/Python/lldbsuite/test/lldbutil.py [hunk 1]
```diff
@@ -759,12 +760,17 @@ def run_to_breakpoint_make_target(test, exe_name = "a.out", in_cwd = True):
...
+    if lldbtest_config.inferior_env:
+        test.runCmd('settings set target.env-vars {}'.format(lldbtest_config.inferior_env))
+
     return target
 
 def run_to_breakpoint_do_run(test, target, bkpt, launch_info = None):
     # Launch the process, and do not stop at the entry point.
     if not launch_info:
-        launch_info = lldb.SBLaunchInfo(None)
+        launch_info = target.GetLaunchInfo()
```
Comment (dcci) [inline lldb/packages/Python/lldbsuite/test/lldbutil.py:773]: Why did you change this call?

File: lldb/packages/Python/lldbsuite/test/lldbutil.py [hunk 2]
```diff
@@ -778,7 +784,11 @@ def run_to_breakpoint_do_run(test, target, bkpt, launch_info = None):
     threads = get_threads_stopped_at_breakpoint(
                 process, bkpt)
 
-    test.assertTrue(len(threads) == 1, "Expected 1 thread to stop at breakpoint, %d did."%(len(threads)))
+    buf = '\n'
+    for i in range(launch_info.GetNumEnvironmentEntries()):
+        buf += 'Env: ' + launch_info.GetEnvironmentEntryAtIndex(i) + '\n'
+
+    test.assertTrue(len(threads) == 1, "Expected 1 thread to stop at breakpoint, %d did. Launch info: %s"%(len(threads), buf))
```
Comment (dcci) [inline lldb/packages/Python/lldbsuite/test/lldbutil.py:791]: What is the output in case of failure here? [just curious :)]

**Comment by dcci** [comment]
Thanks for taking care of this, Vedant!

File: lldb/packages/Python/lldbsuite/test/lldbutil.py (hunk 1 above)
Comment (vedantk) [reply lldb/packages/Python/lldbsuite/test/lldbutil.py:773]: To avoid reparsing the `lldbtest_config.inferior_env` string and populating environment vars into a fresh launch info.

File: lldb/packages/Python/lldbsuite/test/lldbutil.py (hunk 2 above)
Comment (vedantk) [reply lldb/packages/Python/lldbsuite/test/lldbutil.py:791]: The old assertion message (expected 1 thread but found 0), followed by a newline-separated list of the environment variables forwarded to the inferior.

File: lldb/packages/Python/lldbsuite/test/lldbutil.py (hunk 1 above)
Comment (dcci) [reply lldb/packages/Python/lldbsuite/test/lldbutil.py:773]: Makes sense.

**Review by dcci — APPROVED** [review]
LGTM
